telemetry_core/build.gradle.kts

```diff
@@ -63,9 +63,24 @@ tasks {
         relocate("com.google.gson", "com.newrelic.relocated")
         minimize()
     }
+    val propertiesDir = "build/generated/properties"
+    val versionFilename = "telemetry.sdk.version.properties"
+    sourceSets.get("main").output.dir(mapOf("builtBy" to "generateVersionResource"), propertiesDir)
+    register("generateVersionResource") {
+        outputs.file(File("$propertiesDir/$versionFilename"))
+        doLast {
+            val folder = file(propertiesDir)
+            folder.mkdirs()
+            val propertiesFile = File(folder.getAbsolutePath(), versionFilename)
+            propertiesFile.writeText("telemetry.sdk.version=${project.version}")
+        }
+    }
+
     build {
         dependsOn(shadowJar)
+        dependsOn("generateVersionResource")
```

Comment:
does the order matter here? Just want to make sure that the shadow jar has the version file in it.

Reply:
It might.  Lemme fiddle with it and fix the formatting.

Reply:
I tried several different build sequences and couldn't get it to NOT generate and include the properties file, so I think it's ok this way.